pkg/dns/aws/dns.go

```diff
@@ -101,13 +101,26 @@ func NewProvider(config Config, operatorReleaseVersion string) (*Provider, error
 	if len(region) == 0 {
 		return nil, fmt.Errorf("region is required")
 	}
+	awsconfig := &aws.Config{}
+	awsconfig.Region = &region
+
+	// TODO: This API will only return hostedzone resources (which are global)
+	// when the region is forced to us-east-1. We don't yet understand why.
+	tagRegion := "us-east-1"
+
+	//If the region is in aws china, cn-north-1 or cn-northwest-1, we should:
+	// 1. hard code route53 api endpoint to https://api.route53.cn, as route53 is not GA in AWS China, and aws sdk didn't have the endpoint.
+	// 2. use the aws china regions to setup tagging api correctly in stead of "use-east-1"
+	if (region == "cn-north-1") || (region == "cn-northwest-1") {
+		endpoint := "https://route53.amazonaws.com.cn"
+		awsconfig.Endpoint = &endpoint
+		tagRegion = region
```

Comment:
for all public regions we use `us-east-1`

can you verify that the resourcetaggingapi returns results for hosted zone for both regions north-1 and northwest-1 or do we have to pick one.

if you have access to aws china maybe add the aws cli invocation results?

Reply:
Good catch, just confirmed with AWS SA, this service hosted in cn-northwest-1 region:
```
Failed when using cn-north-1:
aws route53  list-hosted-zones  --region cn-north-1 --endpoint-url https://route53.amazonaws.com.cn 
aws route53   list-tags-for-resources  --region cn-north-1 --endpoint-url https://route53.amazonaws.com.cn --resource-id <id> --resource-type hostedzone
Succeed with cn-northwest-1:
aws route53  list-hosted-zones  --region cn-northwest-1 --endpoint-url https://route53.amazonaws.com.cn --> succeed
aws route53   list-tags-for-resources  --region cn-northwest-1 --endpoint-url https://route53.amazonaws.com.cn --resource-id <id> --resource-type hostedzone
```